ClassLibrary/ClassLibrary.Test/TestLibrary.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using NUnit.Framework;
+using Shouldly;
+
+namespace ClassLibrary.Test
+{
+    [TestFixture]
+    class TestLibrary
+    {
+
+        [Test]
+        public void BorrowBooks()
+        {
+            Library lib = new Library();
+            //Add Book
+            Author author = new Author("Lev", "Tolstoj");
+            Book book = new Book("226611156", "War and Peace", author);
+            lib.AddBookInLibrary(book);
+            //Add reader
+            Reader reader = new Reader("Ivan", "Ivanov");
+            lib.AddReader(reader);
+            //Borrow Books
+            lib.ReaderBoroweBook(reader.ID, book.NameBook);
+
+            lib.GetReader(reader.ID).BorrowedBooks.Count.ShouldBe(1);
+            lib.GetReader(reader.ID).BorrowedBooks[0].ShouldBe(book);
+
+        }
+
+        [Test]
+        public void NotReader()
+        {
+            Library lib = new Library();
+            //Add Book
+            Author author = new Author("Lev", "Tolstoj");
+            Book book = new Book("226611156", "War and Peace", author);
+            lib.AddBookInLibrary(book);
+
+            Reader reader = new Reader("Ivan", "Ivanov");
+            Should.Throw<ArgumentException>(() => lib.ReaderBoroweBook(reader.ID, book.NameBook)).Message.ShouldBe($"Not Exist Reader with (ID={reader.ID}) in DepartmentReaders");
+        }
+
+        [Test]
+        public void NotSetNamebook()
+        {
+            Library lib = new Library();
+            //Add Book
+            Author author = new Author("Lev", "Tolstoj");
+            Book book = new Book("226611156", "War and Peace", author);
+            //Add reader
+            Reader reader = new Reader("Ivan", "Ivanov");
+            lib.AddReader(reader);
+
+            Should.Throw<ArgumentNullException>(() => lib.ReaderBoroweBook(reader.ID, "")).Message.ShouldBe("Not set Namebook");
+        }
+
+        [Test]
+        public void NotThisBooK()
+        {
+            Library lib = new Library();
+            //Add Book
+            Author author = new Author("Lev", "Tolstoj");
+            Book book = new Book("226611156", "War and Peace", author);
+
+            //Add reader
+            Reader reader = new Reader("Ivan", "Ivanov");
+            lib.AddReader(reader);
+
+            Should.Throw<ArgumentException>(() => lib.ReaderBoroweBook(reader.ID, book.NameBook)).Message.ShouldBe($"This book ({book.NameBook}) not exist in RepositoryBooks");
+        }
+
+        [Test]
+        public void ALLBooKWithISBNBorrowed()
```

Comment:
Please add a test (I think it was here before) for the case "reader tries to borrow the same book twice".

Reply:
yes, it was here before in TestEnity, but was delete